client/apis/volumegroupsnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+	DeletionPolicy *snapshotv1.DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,4,opt,name=deletionPolicy"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotClassList is a collection of VolumeGroupSnapshotClasses.
+// +kubebuilder:object:root=true
+type VolumeGroupSnapshotClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is the list of VolumeGroupSnapshotClass
+	Items []VolumeGroupSnapshotClass `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotContent represents the actual "on-disk" group snapshot object
+// in the underlying storage system
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster,shortName=vsc;vgscs
+// +kubebuilder:subresource:status
+type VolumeGroupSnapshotContent struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   VolumeGroupSnapshotContentSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status VolumeGroupSnapshotContentStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotContentList is a list of VolumeGroupSnapshotContent objects
+// +kubebuilder:object:root=true
+type VolumeGroupSnapshotContentList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items is the list of VolumeGroupSnapshotContent
+	Items []VolumeGroupSnapshotContent `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// VolumeGroupSnapshotContentSpec describes the common attributes of a group snapshot content
+type VolumeGroupSnapshotContentSpec struct {
+	// VolumeGroupSnapshotRef specifies the VolumeGroupSnapshot object to which this
+	// VolumeGroupSnapshotContent object is bound.
+	// VolumeGroupSnapshot.Spec.VolumeGroupSnapshotContentName field must reference to
+	// this VolumeGroupSnapshotContent's name for the bidirectional binding to be valid.
+	// For a pre-existing VolumeGroupSnapshotContent object, name and namespace of the
+	// VolumeGroupSnapshot object MUST be provided for binding to happen.
+	// This field is immutable after creation.
+	// Required.
+	VolumeGroupSnapshotRef core_v1.ObjectReference `json:"volumeGroupSnapshotRef" protobuf:"bytes,1,opt,name=volumeGroupSnapshotRef"`
+
+	// DeletionPolicy determines whether this VolumeGroupSnapshotContent and the
+	// physical snapshots on the underlying storage system should be deleted when
+	// the bound VolumeGroupSnapshot is deleted.
+	// Supported values are "Retain" and "Delete".
+	// For dynamically provisioned group snapshots, this field will automatically
```

Comment:
// "Retain" means that the VolumeGroupSnapshotContent and its physical group snapshot on underlying storage system are kept.
	// "Delete" means that the VolumeGroupSnapshotContent and its physical group snapshot on underlying storage system are deleted.